modules/core/src/main/scala/predicate.scala

```diff
@@ -33,6 +33,16 @@ trait Term[T] extends Product with Serializable { // fun fact: making this covar
 
   def forall(f: Term[_] => Boolean): Boolean =
     f(this) && children.forall(_.forall(f))
+
+  def map[U](f: T => U): Term[U] =
```

Comment:
Unfortunately this isn't going to work at all. The SQL compiler depends on being able to pattern match on terms to translate query algebra predicates in SQL conditions and this map completely obscures the underlying operation. We need to find some other way of representing optionality which is accessible to the pattern matcher.